static/js/components/ClassificationForm.jsx

```diff
@@ -0,0 +1,207 @@
+  const latest_taxonomyList = taxonomyList.filter((t) => t.isLatest);
+
+  function reducer(state, action) {
+    switch (action.name) {
+      case 'taxonomy_index':
+        return {
+          ...state,
+          [action.name]: action.value,
+          allowed_classes: latest_taxonomyList[action.value].allowed_classes,
+        };
+      default:
+        return {
+          ...state,
+          [action.name]: action.value
+        };
+    }
+  }
+
+  const submitDispatch = useDispatch();
+
+  const initialState = {
+    taxonomy_index: latest_taxonomyList.length > 0 ? 0 : null,
+    classification: null,
+    probability: 1.0,
+    class_select_enabled: false,
+    probability_select_enabled: false,
+    probability_errored: false,
+    allowed_classes: latest_taxonomyList.length > 0 ?
+      latest_taxonomyList[0].allowed_classes : [null]
+  };
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const { handleSubmit } = useForm();
+
+  const useStyles = makeStyles((theme) => ({
+    formControl: {
+      margin: theme.spacing(1),
+      fullWidth: true,
+      display: 'flex',
+      wrap: 'nowrap'
+    }
+  }));
+  const classes = useStyles();
+
+  if (latest_taxonomyList.length === 0) {
+    return (
+      <b>
+        No taxonomies loaded...
+      </b>
+    );
+  }
+
+  const handleTaxonomyChange = (event) => {
+    dispatch({ name: "taxonomy_index", value: event.target.value });
+    dispatch({ name: "classification", value: "" });
+    dispatch({ name: "class_select_enabled", value: true });
+    dispatch({ name: "probability_select_enabled", value: false });
+    dispatch({ name: "probability_errored", value: false });
+    dispatch({ name: "probability", value: 1.0 });
+  };
+
+  const handleClasschange = (event, value) => {
+    dispatch({ name: "classification", value });
+    dispatch({ name: "probability_select_enabled", value: true });
+  };
+
+  const processProb = (event) => {
+    // make sure that the probability in in [0,1], otherwise set
+    // an error state on the entry
+    if ((Number.isNaN(parseFloat(event.target.value))) ||
+       ((parseFloat(event.target.value) > 1) ||
+       (parseFloat(event.target.value) < 0))) {
+      dispatch({ name: "probability_errored", value: true });
+    } else {
+      dispatch({ name: "probability_errored", value: false });
+      dispatch({ name: "probability", value: event.target.value });
+    }
+  };
+
+  const onSubmit = () => {
+    // TODO: allow fine-grained user groups in submission
+    const formData = {
+      taxonomy_id: latest_taxonomyList[state.taxonomy_index].id,
+      obj_id,
+      classification: state.classification.class,
+      probability: parseFloat(state.probability)
+    };
+    // We need to include this field in request, but it isn't in form data
+    submitDispatch(Actions.addClassification(formData));
+  };
+
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
```suggestion
      <form onSubmit={onSubmit}>
```
after removing react-hook-form instantiation above, which is entirely unused except for here

Reply:
done.

Reply:
If you remove this form onSubmit part altogether, change the button type below to "button" and set its `onClick={onSubmit}`, you can remove RHF altogether as no inputs are actually using it

Reply:
Sorry to keep insisting on this, but I really don't think we should be using RHF just to trigger our submit function without registering any components, etc, with it.

Reply:
It just adds an unnecessary layer of complexity, which will slow us down in the long run